index.js

```diff
@@ -1,66 +1,190 @@
+        },
     },
-    "parserOptions": {
-        "ecmaVersion": 2022,
+    // Configuration for TypeScript files only.
+    // Use the recommended TypeScript configs as a base, follow up with our overrides.
+    ...typescriptEslint.configs.recommended.map((conf) => ({
+        ...conf,
+
+        files: ['**/*.ts', '**/*.tsx', '**/*.mts', '**/*.cts'],
+    })),
+    {
+        files: ['**/*.ts', '**/*.tsx', '**/*.mts', '**/*.cts'],
+
+        plugins: {
+            '@typescript-eslint': typescriptEslint.plugin,
+        },
+
+        languageOptions: {
+            parser: typescriptEslint.parser,
+            ecmaVersion: 2022,
+        },
+
+        rules: {
+            // TypeScript doesn't support imports with `.ts` extensions, and `.js` makes the rule complain.
+            'import/extensions': 'off',
+            // Allow semicolons at the end of the code block.
+            '@typescript-eslint/no-extra-semi': 'off',
+            // Note: you must disable the base rule as it can report incorrect errors.
+            'no-shadow': 'off',
+            // This rule extends the base eslint/no-shadow rule. It adds support for TypeScript's this parameters and global augmentation, and adds options for TypeScript features.
+            '@typescript-eslint/no-shadow': ['error'],
+            // Not allowing non-null assertions is overly strict.
+            '@typescript-eslint/no-non-null-assertion': 'off',
+            // Note: you must disable the base rule as it can report incorrect errors.
+            'no-useless-constructor': 'off',
+            // This rule extends the base eslint/no-useless-constructor rule for TypeScript.
+            '@typescript-eslint/no-useless-constructor': 'error',
+            // Note: you must disable the base rule as it can report incorrect errors.
+            'default-param-last': 'off',
+            // This rule extends the base eslint/default-param-last rule for TypeScript.
+            '@typescript-eslint/default-param-last': 'error',
+            // Force promises to be awaited, returned, voided, .then()ed or .catch()ed.
+            '@typescript-eslint/no-floating-promises': 'error',
+            // Disallow the use of promises in places where they shouldn't be used.
+            '@typescript-eslint/await-thenable': 'error',
+            // Do not misuse promises in places like if conditions, or passing them to functions that don't handle promises.
+            '@typescript-eslint/no-misused-promises': ['error', {
+                checksVoidReturn: {
+                    arguments: false,
+                    attributes: false,
+                },
+            }],
+            // Require methods and fucnctions to be marked async if they return a Promise.
+            '@typescript-eslint/promise-function-async': 'error',
+            // Force the usage of "import type" for type imports.
+            '@typescript-eslint/consistent-type-imports': 'error',
```

Comment:
Seems this defaults to `fixStyle: 'separate-type-imports'` which would error on `import { type CheerioCrawlingContext }` if I'm reading it right? We use that syntax in some places but I'm ok to be overturned :)

https://typescript-eslint.io/rules/consistent-type-imports/#fixstyle

Reply:
I hate `import { type` it feels messy and ends up having several `type`s mixed between the imports as the file grows. Also apparently this is better performance 🤔

Reply:
I don't care, but having only one type seems good. I guess this can also be autofixed.